flixel/addons/weapon/FlxWeapon.hx

```diff
@@ -51,106 +51,126 @@ class FlxTypedWeapon<TBullet:FlxBullet>
+	public var useParentDirection(default, set):Bool;
 
 	/**
-	 * If parent is null, the Weapon will fire from a fixed x/y position on the screen, like in the game Missile Command.
+	 * A fixed position from which to fire the weapon, like in the game Missile Command. Only accessible when `fireFrom == POSITION`.
 	 */
 	public var firePosition(default, null):FlxBounds<FlxPoint>;
 
 	/**
-	 * When the bullet is fired if you need to offset it on the x/y axis, for example to line it up with the "nose" of a space ship, set the amount here (positive or negative)
+	 * A value used to offset a bullet's position when it is fired. Can be used to, for example, line a bullet up with the "nose" of a space ship.
+	 * Only accessible when `fireFrom == PARENT`.
 	 */
+	@:deprecated("Use positionOffsetBounds instead")
 	public var positionOffset(default, null):FlxPoint;
 
+	/**
+	 * A value used to offset a bullet's position when it is fired. Can be used to, for example, line a bullet up with the "nose" of a space ship.
+	 * Only accessible when `fireFrom == PARENT`.
+	 */
+	public var positionOffsetBounds(default, null):FlxBounds<FlxPoint>;
+
 	public var fireFrom(default, set):FlxWeaponFireFrom;
 	public var speedMode:FlxWeaponSpeedMode;
 
 	/**
 	 * The lifespan of the bullet, given in seconds.
-	 * The bullet will be killed once it passes this lifespan, if still alive and in bounds.
+	 * The bullet will be killed once it passes this lifespan, if it is still alive and in bounds.
 	 */
 	public var bulletLifeSpan:FlxBounds<Float>;
 
 	/**
 	 * The elasticity of the fired bullet controls how much it rebounds off collision surfaces.
-	 * Between 0 and 1 (0 being no rebound, 1 being 100% force rebound). Set to zero to disable.
+	 * Between `0` and `1` (`0` being no rebound, and `1` being 100% force rebound). Default is `0`.
 	 */
 	public var bulletElasticity:Float = 0;
 
 	/**
-	 * Auto set the bullet angle when firing, such that it faces towards the target
+	 * Whether to automatically set the bullet's angle when firing, such that it faces towards the target.
 	 */
 	public var rotateBulletTowardsTarget:Bool = false;
 
 	/**
-	 * A reference to the Bullet that was last fired
+	 * A reference to the bullet that was last fired.
 	 */
 	public var currentBullet:TBullet;
 
-	// Callbacks
+	/**
+	 * A function that is called immediately before a bullet is fired.
+	 */
 	public var onPreFireCallback:Void->Void;
+
+	/**
+	 * A function that is called immediately after a bullet is fired.
+	 */
 	public var onPostFireCallback:Void->Void;
 
-	// Sounds
+	/**
+	 * A sound that is played immediately before a bullet is fired.
+	 */
 	public var onPreFireSound:FlxSound;
+
+	/**
+	 * A sound that is played immediately after a bullet is fired.
+	 */
 	public var onPostFireSound:FlxSound;
 
 	/**
-	 * The factory function to create a bullet
+	 * The factory function used to create new bullets.
 	 */
 	var bulletFactory:FlxTypedWeapon<TBullet>->TBullet;
 
-	var lastFired:Int = 0;
```

Comment:
removing this is technically a breaking change but i'll allow it. In all likelyhood no real person will have a compile error unless they use this field in an extending class, and highly unlikely and easy to fix